src/routes/transactions/mappers/common/safe-app-info.mapper.ts

```diff
@@ -0,0 +1,52 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { SafeAppsRepository } from '../../../../domain/safe-apps/safe-apps.repository';
+import { ISafeAppsRepository } from '../../../../domain/safe-apps/safe-apps.repository.interface';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { SafeAppInfo } from '../../entities/safe-app-info.entity';
+
+@Injectable()
+export class SafeAppInfoMapper {
+  private readonly logger = new Logger(SafeAppInfoMapper.name);
+  private static readonly IPFS_URL = 'ipfs.io';
+  private static readonly CF_IPFS_URL = 'cloudflare-ipfs.com';
+
+  constructor(
+    @Inject(ISafeAppsRepository)
+    private readonly safeAppsRepository: SafeAppsRepository,
+  ) {}
+
+  async mapSafeAppInfo(
+    chainId: string,
+    transaction: MultisigTransaction,
+  ): Promise<SafeAppInfo | null> {
+    const originUrl = this.getOriginUrl(transaction);
+    if (!originUrl) return null;
+
+    const [safeApp] = await this.safeAppsRepository.getSafeApps(
+      chainId,
+      undefined,
+      originUrl,
+    );
+    if (!safeApp) {
+      this.logger.error(
+        `No Safe Apps matching the origin url ${originUrl} (safeTxHash: ${transaction.safeTxHash})`,
+      );
+      return null;
+    }
+
+    return new SafeAppInfo(
+      safeApp.name,
+      safeApp.url.replace(
+        SafeAppInfoMapper.IPFS_URL,
+        SafeAppInfoMapper.CF_IPFS_URL,
+      ),
+      safeApp.iconUrl,
+    );
+  }
+
+  private getOriginUrl(transaction: MultisigTransaction): string | null {
+    return transaction.origin
+      ? JSON.parse(transaction.origin).url ?? null
+      : null;
+  }
```

Comment:
If we expect this to be a JSON payload with a `url` why not solve this at the AJV level?

Reply:
Hmm, `transaction.origin` is actually a `string` which contains a JSON. By solving it at AJV level you mean using a custom keyword as we did for `isDate`? I'm saying this because I think we would need to parse the JSON content explicitly to validate it 🤔

Reply:
I meant if we expect it to be of a JSON format always then we can try to solve it at the parsing
/validation level but I get your point. Let's leave it like this for now 👍